client.go

```diff
@@ -1229,3 +1205,27 @@ func nodePathToURL(node *Node, path string) url.URL {
 		Path:   path,
 	}
 }
+
+type InternalClient interface {
```

Comment:
I just want to leave a note here for discussion: typically an interface containing more than two or three methods is an indication that something hasn't been architected correctly. While I understand that putting the entire Client behind an interface might require this many methods, I don't want to let it go unquestioned.

Reply:
Would it be OK if I added your comment to the interface and we would address that in another PR?

Reply:
ok with me.